x-pack/legacy/plugins/siem/server/lib/detection_engine/alerts/utils.test.ts

```diff
@@ -0,0 +1,658 @@
+          name: sampleParams.name,
+          severity: sampleParams.severity,
+          description: sampleParams.description,
+          original_time: sampleDocNoSortId._source['@timestamp'],
+          index_patterns: sampleParams.index,
+          references: sampleParams.references,
+        },
+      });
+    });
+  });
+  describe('singleBulkIndex', () => {
+    test('create successful bulk index', async () => {
+      const sampleParams = sampleSignalAlertParams(undefined);
+      const sampleSearchResult = sampleDocSearchResultsNoSortId;
+      const savedObjectsClient = savedObjectsClientMock.create();
+      const bulkBody = sampleDocSearchResultsNoSortId.hits.hits.flatMap(doc => [
+        {
+          index: {
+            _index: process.env.SIGNALS_INDEX || '.siem-signals-10-01-2019',
+            _id: doc._id,
+          },
+        },
+        buildBulkBody(doc, sampleParams),
+      ]);
+      const mockService: AlertServices = {
+        callCluster: async (action: string, params: BulkIndexDocumentsParams) => {
+          expect(action).toEqual('bulk');
+          expect(params.index).toEqual('.siem-signals-10-01-2019');
+
+          // timestamps are annoying...
+          (bulkBody[1] as SignalHit).signal['@timestamp'] = params.body[1].signal['@timestamp'];
+          expect(params.body).toEqual(bulkBody);
+          return {
+            took: 100,
+            errors: false,
+            items: [
+              {
+                fakeItemValue: 'fakeItemKey',
+              },
+            ],
+          };
+        },
+        alertInstanceFactory: jest.fn(),
+        savedObjectsClient,
+      };
+      const successfulSingleBulkIndex = await singleBulkIndex(
+        sampleSearchResult,
+        sampleParams,
+        mockService,
+        mockLogger
+      );
+      expect(mockLogger.debug).toHaveBeenCalledTimes(2);
+      expect(mockLogger.warn).toHaveBeenCalledTimes(0);
+      expect(successfulSingleBulkIndex).toEqual(true);
+    });
+    test('create unsuccessful bulk index due to empty search results', async () => {
+      const sampleParams = sampleSignalAlertParams(undefined);
+      const sampleSearchResult = sampleEmptyDocSearchResults;
+      const savedObjectsClient = savedObjectsClientMock.create();
+      const mockService: AlertServices = {
+        callCluster: async (action: string, params: BulkIndexDocumentsParams) => {
+          expect(action).toEqual('bulk');
+          expect(params.index).toEqual('.siem-signals-10-01-2019');
+          return false; // value is irrelevant
+        },
+        alertInstanceFactory: jest.fn(),
+        savedObjectsClient,
+      };
+      const successfulSingleBulkIndex = await singleBulkIndex(
+        sampleSearchResult,
+        sampleParams,
+        mockService,
+        mockLogger
+      );
+      expect(mockLogger.debug).toHaveBeenCalledTimes(0);
+      expect(mockLogger.error).toHaveBeenCalledTimes(0);
+      expect(successfulSingleBulkIndex).toEqual(true);
```

Comment:
Having the return value being checked is great though 👍! I like the tests around the return types.